src/mscorlib/src/System/Collections/Generic/Dictionary.cs

```diff
@@ -515,12 +507,26 @@ private struct Entry {
             return false;
         }
 
-        public bool TryGetValue(TKey key, out TValue value) {
-            int i = FindEntry(key);
-            if (i >= 0) {
-                value = entries[i].value;
-                return true;
+        public bool TryGetValue(TKey key, out TValue value)
+        {
+            if (key == null)
```

Comment:
Don't know, see the `default(T) == null` test in quite a few places

https://github.com/dotnet/coreclr/blob/master/src/mscorlib/src/System/ThrowHelper.cs#L113-L116

https://github.com/dotnet/coreclr/blob/master/src/mscorlib/src/System/Collections/ObjectModel/Collection.cs#L322-L325

https://github.com/dotnet/coreclr/blob/master/src/mscorlib/src/System/Collections/ObjectModel/ReadOnlyCollection.cs#L200-L204